22.md

```diff
@@ -0,0 +1,366 @@
+This NIP introduces the concept of Smart Widgets for Nostr clients. Smart Widgets are customizable, interactive components that can be embedded within Nostr events. These widgets allow users and developers to create rich, dynamic content that enhances the user experience. This proposal outlines the structure, implementation, and potential use cases of Smart Widgets within the Nostr protocol.
+
+# 1. Motivation
+As Nostr continues to grow, the need for more engaging and interactive content increases. Traditional static content limits user interaction and engagement. Smart Widgets provide a way to embed dynamic components, such as buttons, images, and polls, directly into Nostr events, enabling more interactive and responsive content.
+
+# 2. Specification
+## 2.1. Event kind 
+A new addressable event kind is proposed for Smart Widgets kind `30031`. This event kind will encapsulate the data necessary to render and interact with the Smart Widgets.
+## 2.2. Event Structure 
+The `content` of these events should be a stringified JSON of the related smart widget. It is required and can't be empty.
+
+The list of tags are as follows:
+- `d` (required) universally unique identifier (UUID). Generated by the client creating the smart widget event.
+- `title` (optional) title of the smart widget. 
+- `summary` (optional) description of the smart widget and its usage.
+- `client` (optional) reference for the used client for creating the smart widget.
+- `published_at` (optional) Publishing time of the smart widget differs from the `created_at` for edited smart widgets.
+
+```JSON
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": 30031,
+  "content": "<smart_widget_stringified_json>",
+  "tags": [
+  
+    ["d", "<UUID>"],
+	
+    ["title", "<title of smart widget event>"],
+    ["summary", "<description of smart widget event>"],
+    
+    ["client", "<client_name>", "31990:app1_pubkey:<d_identifier>"],
+    ["published_at", "<Unix timestamp in seconds>"],
+    
+  ]
+}
+```
+## 2.3. Positional Key Elements:
+The `content` property should contain a stringified object that defines the structure of the smart widget, it contains three main properties:
+```JSON
+{
+  "background_color": "<string_hex_color_code>", 
+  "border_color": "<string_hex_color_code> | <empty_string>", // leave empty if not specified
+  "components": "<array_of_objects>"
+}
+```
+### 2.3.1. background_color
+the `background_color` property defines the color of the overall container of the smart widget.
```

Comment:
with or without alpha?

Reply:
It's a hex code, so yes, it can contain alpha.